wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,336 @@
+/**
+	DamageModify
+
+	Never use both this and DamageType package.
+	
+	API:
+
+	// Essentially used for callback closures taking on the form Damage -> null
+	interface ForDamageCallback
+		run(Damage d)
+
+	// These adjust effective damage before any has occurred.
+	// Can be called zero or more times.
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
+	// are called strictly once per damage instance
+	addDamageListener(ForDamageCallback cb)
+	removeDamageListener(ForDamageCallback cb)
+
+	// a helper for 'new Damage(this, target, amount, isSpell, id)..modify()..deal()'
+	unit.damage(unit target, real amount, OPTIONAL bool isSpell = true, int id = 0)
+
+	// a helper for '= new Damage(this, target, amount, isSpell, id)..saveInstance()..modify()..deal()'
+	unit.returnDamage(unit target, real amount, bool isSpell, int id) -> Damage
+
+	class Damage
```

Comment:
This "vjass style" doc of copying the entire class into a comment for documentation is frankly just terrible.
Document the actual class with hotdoc and just provide valuable information in the package doc on how to get started immediately plus most common usages. Most of the comments here are absolutely superfluous and don't add anything of value.